openff/bespokefit/cli/executor/submit.py

```diff
@@ -156,67 +160,135 @@ def _to_input_schema(
 
 def _submit(
     console: "rich.Console",
-    input_file_path: Optional[str],
-    molecule_smiles: Optional[str],
+    input_file_path: Optional[List[str]],
+    molecule_smiles: Optional[List[str]],
     force_field_path: Optional[str],
     target_torsion_smirks: Tuple[str],
     workflow_name: Optional[str],
     workflow_file_name: Optional[str],
-) -> str:
+    allow_multiple_molecules: bool,
+    save_submission: bool,
+) -> List[str]:
 
     from openff.toolkit.topology import Molecule
 
     from openff.bespokefit.executor import BespokeExecutor
 
     console.print(Padding("1. preparing the bespoke workflow", (0, 0, 1, 0)))
 
-    if input_file_path is not None:
+    if input_file_path:
         with console.status("loading the molecules"):
-            molecule = Molecule.from_file(input_file_path)
-
+            all_molecules = []
+            for input_file in input_file_path:
+                file_molecules = Molecule.from_file(input_file)
+                if isinstance(file_molecules, Molecule):
+                    file_molecules.properties["input_file"] = input_file
+                    all_molecules.append(file_molecules)
+                else:
+                    for m in file_molecules:
+                        m.properties["input_file"] = input_file
+                    all_molecules.extend(file_molecules)
     else:
         with console.status("creating molecule from smiles"):
-            molecule = Molecule.from_smiles(molecule_smiles)
+            all_molecules = [Molecule.from_smiles(smiles) for smiles in molecule_smiles]
 
-    if not isinstance(molecule, Molecule) or "." in molecule.to_smiles():
+    if not allow_multiple_molecules and len(all_molecules) > 1:
 
         exit_with_messages(
-            "[[red]ERROR[/red]] only one molecule can currently be submitted at "
-            "a time",
+            "[[red]ERROR[/red] only one molecule can be submitted at once",
             console=console,
             exit_code=2,
         )
 
-    console.print("[[green]✓[/green]] [blue]1[/blue] molecule was found")
+    for molecule in all_molecules:
+        if "." in molecule.to_smiles():
+
+            exit_with_messages(
+                f"[[red]ERROR[/red]] complexes are not supported, {molecule} can not be submitted!",
+                console=console,
+                exit_code=2,
+            )
+
+    console.print(
+        f"[[green]✓[/green]] [blue]{len(all_molecules)}[/blue] molecules found"
+    )
 
-    with console.status("building fitting schemas"):
+    input_schemas = []
+    for molecule in track(
+        all_molecules,
+        description="building fitting schemas",
+        console=console,
+        total=len(all_molecules),
```

Comment:
```suggestion
        total=len(all_molecules),
        transient=True,
```

It might be nice to hide the progress bar after it gets to the end as we show a green tick on the next line, and I think(?) adding `transient=True` should do this?

Reply:
Nice that would look good Ill try it out!